### ISSUE TITLE
Associations use :create strategy when parent strategy is :build

### ISSUE THREAD

**Issue by tmertens** [thoughtbot/factory_girl#702]
TLDR: **Why do associations always use the `:create` build strategy rather than defaulting to the strategy used for the parent/calling factory?**

This is half a question and half a real issue.  I haven't created a PR yet because I wanted to discuss the issue first.

When explicitly defining associations and/or using the default association mechanism in FactoryGirl, if the `:build` strategy is not explicitly specified for an association, FactoryGirl will _always_ `:create` the association, even when the parent factory is called by the user via the `:build` strategy.

This produced real failures in (system integration) tests because associations were actually created rather than only being built.  To make a long story short, it resulted in uniqueness violations when the tests were executed, causing the tests to fail.

After some experimentation, I found that using `:build` for associations seems to work just as well as `:create`, without persisting the associated model to the database, while the same traits which build associations also correctly persist the associated records on `:create` of the parent factory.

Given:

```ruby
FactoryGirl.define do
  factory :user do
    sequence(:email) { |n| "person#{n}@example.test" }

    ### A default association, will :create a Person on :build of the user factory.
    person
    ### Will :build an Address on :build of the user factory.
    association :address, strategy: :build

   ### A strategy to build has_many associations on build or create, associations are persisted successfully on :create of the parent factory
    trait :posts do
      posts { build_list(:post, 1) }
    end
  end
end

user = FactoryGirl.build(:user)
user.posts.count #=> 1
User.count #=> 0 
Post.count #=> 0 because we used build_list in the trait block
Person.count #=> 1 because FactoryGirl defaults to :create rather than :build by default for assocations

## Now, suppose we have a uniqueness constraint on Person.
## Building another user would fail, even though we didn't call create on the :user factory, nor save the resulting record:
FactoryGirl.build(:user)
#=> uniqueness error because the Person was already created on building the first :user.
```

Where this became a real issue was when we were using FactoryGirl.build to generate data for filling forms for front-end testing, which would result in persisted associations and thus issues related to record/data uniqueness when submitting data from the front-end.

We're presently working around this issue in our factories by explicitly declaring the build strategy for associations, but the question remains why `:create` is the default, rather than the strategy used for the parent/calling factory.

**Comment by joshuaclayton** [comment]
Closing due to inactivity.
